doc/interfaces/registration_flow.rst

```diff
@@ -0,0 +1,116 @@
+
+    dipy_align_affine <path_to_static_file> <path_to_moving_file> --out_dir "affine_reg_output" --out_affine "affine_reg.txt"
+
+This command will apply affine transformation on the moving image file and save the
+transformed image and the affine matrix to the directory ``affine_reg_output``.
+
+In case you did not specify the output directory, the transformed image file and 
+affine matrix would be saved in the input directory by default. If you did not
+specify the name of the output affine matrix, the affine matrix will be saved to a
+file named ``affine.txt`` by default, located in the output directory also by
+default.
+
+------------------------------------
+Symmetric Diffeomorphic Registration
+------------------------------------
+
+Symmetric Diffeomorphic Registration is performed using the ``Symmetric Normalization
+(SyN)`` algorithm. It is an optimization technique that brings the moving image closer
+to the static image.
+
+Create a directory in which to save the transformed image (e.g.: ``syn_reg_output``)::
+    
+    mkdir syn_reg_output
+
+To run the symmetric normalization registration method, we need to specify the 
+paths to the static image file, and to the moving image file, followed by
+optional arguments. In this case, we will be specifying the metric (``metric``), the
+output directory (``out_dir``) and the file name of the output warped image
+(``out_warped``). You can use cc (Cross Correlation), ssd (Sum Squared Difference)
+or em (Expectation-Maximization) as metrics.
+
+We will run the command as::
+
+    dipy_align_syn <path_to_static_file> <path_to_moving_file> --metric "cc" --out_dir "syn_reg_output" --out_warped "syn_reg_warped.nii.gz"
+
+This command will perform symmetric diffeomorphic registration and save it to 
+the specified output directory.
+
+In case you did not specify the output directory, the transformed files would be 
+saved in the input directory by deafult. If you did not specify the file name of
+the output warped image, the warped file will be saved as ``warped_moved.nii.gz``
+by default.
+
+----------------------
+Apply Registration
+----------------------
+
+In order to apply transform registration, we need to specify the path of the 
+static image file, moving image file, and transform map file followed by 
+optional arguments. In this case, we will be specifying the transform type
+(``transform_type``) and the output directory (``out_dir``).
+
+Create a directory in which to save the transfomed files (e.g.: ``transform_reg_output``)::
+
+    mkdir transform_reg_output
+
+Then, run the command as::
+
+    dipy_apply_transform <path_to_static_file> <path_to_moving_file> <path_to_transform_map_file> --transform_type "diffeomorphic" --out_dir "transform_reg_output"
+
+This command will transform the moving image and save the transformed files 
+to the specified output directory.
+
+-----------------------------
+Streamline-based Registration
+-----------------------------
```

Comment:
Please add the citation/reference for this method.

Reply:
Addressed in adcad46